### PR TITLE
first pass at submitting analyses; work in progress

### PR DESCRIPTION
**Pull request by davidangb**
This is a very early PR, since I know everybody is waiting on this: it adds a button to submit an analysis. However ...

*there is no feedback to the end user yet*, beyond browser console logging. So it's pretty incomplete, but hopefully I can get early code review, and others can get access to this functionality for their own testing.

Still left to do:
* success/error messaging to the user; something primitive, but we need something.
* allow the user to submit an empty expression, which then will use the rawls/cromwell defaults
* better canned response for development, when not using live data

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 1]
```diff
@@ -195,7 +195,41 @@
                            (m "attributes")])
                      (:entities @state))}]])
         (style/create-form-label "Define Expression")
-        (style/create-text-field {})]])}])
+        (style/create-text-field {:ref "expressionname" :defaultValue "this"})]
+
+       [:div {:style {:fontSize "106%" :lineHeight 1 :textAlign "center"}}
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: If you use Matt's Button class (in components), you'll get nice button styling for free.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 2]
```diff
@@ -195,7 +195,41 @@
...
+        [:div {:style   {:padding         "0.7em 0" :cursor "pointer"
+                         :backgroundColor (:button-blue style/colors)
+                         :color "#fff" :borderRadius 4
+                         :border          (str "1px solid " (:line-gray style/colors))}
+
+               ;; TODO: what should we show in the UI after submitting?
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: I think a `js/alert` would work for a first pass.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 3]
```diff
@@ -195,7 +195,41 @@
...
+                                                           :entityType                   ((:selected-entity @state) "entityType")
+                                                           :entityName                   ((:selected-entity @state) "name")
+                                                           :expression                   (-> (@refs "expressionname") .getDOMNode .-value)
+                                                           })
+                               :headers {"Content-Type" "application/json"}
+                               :on-done (fn [{:keys [success? xhr]}]
+                                            (if success?
+                                              (utils/rlog "Submission id: " ((utils/parse-json-string (.-responseText xhr)) "submissionId"))
+                                              (utils/rlog "Error: " (.-responseText xhr))))
+                               :canned-response {:responseText (utils/->json-string
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: Seems like all you really need is: `{"submissionId" "my-test-submission"}`

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 4]
```diff
@@ -195,7 +195,41 @@
...
+                         :backgroundColor (:button-blue style/colors)
+                         :color "#fff" :borderRadius 4
+                         :border          (str "1px solid " (:line-gray style/colors))}
+
+               ;; TODO: what should we show in the UI after submitting?
+               ;; TODO: if no expression entered, do not send one in the ajax request
+               ;; TODO: disable/enable submit button based on the state of entity selection & expression value
+               ;; TODO: diable submit button after submitting
+               ;; TODO: better canned response
+               :onClick (fn [{:keys []}]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: This function only takes one argument: the click event. This should be either `(fn [e]...)` or `(fn [] ...)`

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 5]
```diff
@@ -195,7 +195,41 @@
...
+                            (utils/ajax-orch
+                              (paths/submit-method-path workspace)
+                              {:method :post
+                               :data (utils/->json-string {:methodConfigurationNamespace (config "namespace")
+                                                           :methodConfigurationName      (config "name")
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: Nit: Prefer to not line these up. Easy to adjust in IntelliJ (if you didn't do this by hand).

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs [hunk 6]
```diff
@@ -195,7 +195,41 @@
...
+        (style/create-text-field {:ref "expressionname" :defaultValue "this"})]
+
+       [:div {:style {:fontSize "106%" :lineHeight 1 :textAlign "center"}}
+        [:div {:style   {:padding         "0.7em 0" :cursor "pointer"
+                         :backgroundColor (:button-blue style/colors)
+                         :color "#fff" :borderRadius 4
+                         :border          (str "1px solid " (:line-gray style/colors))}
+
+               ;; TODO: what should we show in the UI after submitting?
+               ;; TODO: if no expression entered, do not send one in the ajax request
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: You'll probably want to do:
```clojure
(let [expression (clojure.string/trim (-> (@refs "expressionname") .getDOMNode .-value))]
  ...
  :data .. (merge {:methodConfigurationNamespace...}
                 (when-not (blank? expression) {:expression expression}))
```

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs (hunk 2 above)
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: I have to say that I strongly dislike js alerts. We should look into a more user-friendly pattern.

Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: I'm with you @rushtong. I think we will eventually want to put the user on the "submission status" page for the returned submission ID. We just need some kind of placeholder until that page is created.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs (hunk 1 above)
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: hmm ... I happen to like the full-width button! But it seems nobody else does :)

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs (hunk 2 above)
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_editor.cljs]: I've implemented a pretty hacky placeholder without an alert ... will update the PR momentarily.

**Comment by davidangb** [comment]
ready for re-review, I hope!

**Comment by dmohs** [comment]
:+1: Let's get this in so we can demo. We can follow-up with cleanups.

**Pull request merged**